src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -166,7 +166,7 @@ public bool ContainsAssemblyOrModuleOrDynamic(ISymbol symbol, bool primary)
                          ? intermediateProjects
                          : intermediateProjects.Add((ProjectState, translate));
 
-                    var newState = State.Create(newInProgressCompilation, state.GeneratorDriver, newIntermediateProjects);
+                    var newState = State.Create(newInProgressCompilation, state.GeneratedDocuments, newIntermediateProjects);
```

Comment:
fwiw, i far prefer this.  it's much easier to reason about the states just pointing at raw immutable data, instead of some sort of 'driver' that implies some sort of logic/mutation.

Reply:
So we _may_ go back to the driver once we want to do better incrementality; the idea originally was the driver lets us say "oh this was changed, that was changed" and then it's holding onto trees that don't need to be recomputed. I removed that at least temporarily while @chsienki and I figure out how that's going to look since the interesting bits (document changing) was removed temporarily. But yes, in the mean time I removed it since it was complexity that wasn't doing any good.